src/err/mod.rs

```diff
@@ -299,7 +299,7 @@ impl PyErr {
         if ptype.as_ptr() == PanicException::type_object_raw(py).cast() {
             let msg: String = pvalue
                 .as_ref()
-                .and_then(|obj| obj.extract(py).ok())
+                .map(|obj| obj.as_ref(py).to_string())
```

Comment:
IIRC, we changed the `Display` impl to call into `write_unraisable` if the call to `PyAny::str` fails. Is this appropriate here? Should we call `and_then(|obj| obj.as_ref(py).str().ok())` instead since we have an alternative fallback here?

Reply:
Agreed, great idea.